atomix/cluster/src/test/java/io/atomix/cluster/messaging/impl/NettyMessagingServiceTest.java

```diff
@@ -705,4 +712,48 @@ void testNoRemoteHandlerExceptionEmptyStringValue() {
         .isInstanceOf(MessagingException.NoRemoteHandler.class)
         .withMessage(expectedException.getMessage());
   }
+
+  @EnabledOnOs(OS.LINUX)
+  @RegressionTest("https://github.com/camunda/zeebe/issues/14837")
+  void shouldNotLeakUdpSockets() throws IOException {
+    // given
+    // the configured amount of threads for Netty's epoll transport
+    final var maxConnections = Runtime.getRuntime().availableProcessors() * 2;
+    final var subject = nextSubject();
+    netty2.registerHandler(
+        subject, (address, bytes) -> new byte[0], MoreExecutors.directExecutor());
+
+    // when
+    for (int i = 0; i < maxConnections * 4; i++) {
+      netty1.sendAndReceive(netty2.address(), subject, "hello world".getBytes(), false).join();
```

Comment:
Do you mean why we set `keepAlive=false` here? It's because otherwise our channel pool is limited, and we won't create more connections as we keep reusing the same channels. But we do use transient connections in production sometimes, and channels get recreated over time - but it's easiest to test with just creating a new channel every time here.

Reply:
I was wondering why each request will create an own socket and client.